SFR1.py

```diff
@@ -0,0 +1,228 @@
+        input_path (str): Path to the original FASTQ file.
+
+    Returns:
+        str: Path to the saved file.
+    """
+    if output_filename == '':  # если имя не передано, используем имя входного файла
+        output_filename = os.path.basename(input_path)
+    if not output_filename.endswith(".fastq"):
+        output_filename += ".fastq"
+    output_dir = 'fastq_filtrator_results'  # задаём имя папки для сохранения результата
+    os.makedirs(output_dir, exist_ok=True)  # создаем папку, если она не существует
+    output_path = os.path.join(output_dir, output_filename)  # задаем путь, по которому сохранится файл
+
+    SeqIO.write(filtered_records, output_path, "fastq")
+
+    return output_path
+
+
+class BiologicalSequence(ABC):
+
+    @abstractmethod
+    def __len__(self) -> int:
+        pass
+
+    @abstractmethod
+    def __getitem__(self, index) -> str:
+        pass
+
+    @abstractmethod
+    def __str__(self) -> str:
+        pass
+
+    @abstractmethod
+    def alphabet_is_valid(self) -> bool:
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    def __init__(self, sequence: str):
+        self.sequence = sequence
+
+    def __len__(self) -> int:
+        return len(self.sequence)
+
+    def __getitem__(self, index) -> str:
+        return self.sequence[index]
+
+    def __str__(self) -> str:
+        return self.sequence
+
+    def alphabet_is_valid(self) -> bool:
+        dna_alphabet = set("ATGCatgc")
+        rna_alphabet = set("AGCUagcu")
+        if isinstance(self, DNASequence):
+            return set(self.sequence).issubset(dna_alphabet)
+        if isinstance(self, RNASequence):
+            return set(self.sequence).issubset(rna_alphabet)
+        return False
+
+    def complement(self) -> 'NucleicAcidSequence':
+        complement_pairs = {
+            'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G', 'U': 'A',
+            'a': 't', 't': 'a', 'g': 'c', 'c': 'g', 'u': 'a'
+        }
+        complement_sequence = ''
+        for base in self.sequence:
+            complement_sequence += complement_pairs.get(base, base)
+
+        return type(self)(complement_sequence)
+
+    def gc_content(self) -> float:
+        gc_count = 0
+        for base in self.sequence:
+            if base in {'G', 'g', 'C', 'c'}:
+                gc_count += 1
+        total_bases = len(self.sequence)
+        gc_percentage = gc_count / total_bases
+        return gc_percentage
+
+
+class DNASequence(NucleicAcidSequence):
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+        self.sequence = sequence
+
+    def transcribe(self) -> 'RNASequence':
+        transcribe_sequence = ''
+        for base in self.sequence:
+            if base == 'T':
+                transcribe_sequence += 'U'
+            if base == 't':
+                transcribe_sequence += 'u'
+            else:
+                transcribe_sequence += base
+        return RNASequence(transcribe_sequence)
+
```

Comment:
После проверки if base == 't':,  используется else, который выполнится для всех случаев, кроме когда base == 't'. Это означает, что если base == 'T', код добавит 'U' к transcribe_sequence, а затем выполнит блок else и добавит 'T' ещё раз, что неверно. Надо 
```suggestion
elif base == 't': 
transcribe_sequence += 'u'
```